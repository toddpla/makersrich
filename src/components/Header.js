import React from 'react';
import {NavLink} from 'react-router-dom';
import { connect } from 'react-redux'
import { startLogout } from '../actions/auth'

export const Header = ({ startLogout }) => (
  <header>
      <NavLink to="/game" className='nav-link' activeClassName="is-active"> Game </NavLink>
      <NavLink to="/add-question" className='nav-link' activeClassName="is-active"> Contribute </NavLink>
      <button onClick={startLogout}>Logout</button>
  </header>
)

const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout())
})

export default connect(undefined, mapDispatchToProps)(Header);
