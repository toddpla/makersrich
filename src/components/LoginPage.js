import React from 'react'
import { connect } from 'react-redux'
import { firebase, googleAuthProvider, githubAuthProvider } from '../firebase/firebase'
import { startLogin } from '../actions/auth'

export class LoginPage extends React.Component {

  handleGoogleLogin = () => {
    firebase.auth().signInWithPopup(googleAuthProvider).then((result) => {
      this.props.startLogin(result.user.uid)
    })
  }

  handleGithubLogin = () => {
    firebase.auth().signInWithPopup(githubAuthProvider).then((result) => {
      this.props.startLogin(result.user.uid)
    })
  }

  render() {
    return (
      <div>
        <button onClick={this.handleGoogleLogin}>Google Login</button>
        <button onClick={this.handleGithubLogin}>Github Login</button>
      </div>
    )
  }
}

const mapDispatchToProps = (dispatch) => ({
  startLogin: (uid) => dispatch(startLogin(uid)),
})

export default connect(undefined, mapDispatchToProps)(LoginPage)
