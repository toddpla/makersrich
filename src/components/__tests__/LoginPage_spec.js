import React from 'react'
import { LoginPage } from '../LoginPage'
import renderer from 'react-test-renderer';
import { Header } from '../Header'
import { shallow } from 'enzyme'

let handleGithubLogin = jest.fn()
let handleGoogleLogin = jest.fn()

test('renders correctly', () => {
  const loginPage = renderer.create(<LoginPage />).toJSON();
  expect(loginPage).toMatchSnapshot();
});
