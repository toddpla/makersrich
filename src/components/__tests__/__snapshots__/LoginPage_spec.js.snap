// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    Google Login
  </button>
  <button
    onClick={[Function]}
  >
    Github Login
  </button>
</div>
`;
